src/types/exec_scope.rs

```diff
@@ -0,0 +1,197 @@
+use crate::vm::errors::exec_scope_errors::ExecScopeError;
+use num_bigint::BigInt;
+use std::collections::HashMap;
+
+#[derive(Eq, PartialEq, Debug)]
+pub struct ExecutionScopes {
+    exec_scopes: Vec<HashMap<String, PythonDictValueType>>,
+}
+
+#[derive(Eq, Hash, PartialEq, Debug)]
+pub enum PythonDictValueType {
+    BigInt(BigInt),
+}
+
+impl ExecutionScopes {
+    pub fn new() -> ExecutionScopes {
+        ExecutionScopes {
+            exec_scopes: vec![HashMap::new()],
+        }
+    }
+
+    pub fn enter_scope(&mut self, new_scope_locals: HashMap<String, PythonDictValueType>) {
+        self.exec_scopes.push(new_scope_locals);
+    }
+
+    pub fn exit_scope(&mut self) -> Result<(), ExecScopeError> {
+        if self.exec_scopes.len() == 1 {
+            return Err(ExecScopeError::ExitMainScopeError);
+        }
+        self.exec_scopes.pop();
+
+        Ok(())
+    }
+
+    pub fn get_local_variables(&mut self) -> Option<&mut HashMap<String, PythonDictValueType>> {
+        self.exec_scopes.last_mut()
+    }
+
+    pub fn assign_or_update_variable(&mut self, var_name: String, var_value: PythonDictValueType) {
```

Comment:
Should probably use `&str`s rather than `String`s. This allows passing literals.